elastic/kibana
# [Index Management] Enrich policies: add support for listing, executing and deleting

## Summary

This is the first of a series of PR's that will be merged in the [enrich-policies feature branch](https://github.com/elastic/kibana/tree/feature/enrich-policies). 

This PR creates the barebones for supporting enrich policies in index management, also gives support for listing policies, executing them and deleting them.

Note to reviewers: 
* Copy and doc links will be separately addressed with https://github.com/elastic/kibana/issues/163808
* API integration tests will be separately addressed with https://github.com/elastic/kibana/issues/163807

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Navigate to console and execute the commands listed below in order to create a policy
3. Navigate to `index management` and click on `enrich policies` tab
4. Verify that:
* The policy you just created is listed there
* Reload and search works
* Empy states (loading, error)
* Policy can be correctly executed and a toast is shown
* Policy can be correctly deleted and a toast is shown
* It shows error toast when failing to delete/execute policy (you can force a failure in the server response of the API)


<details>
<summary>Create a policy</summary>

```
PUT /users
{
  "mappings": {
    "properties": {
      "email": {
        "type": "keyword"
      },
      "first_name": {
        "type": "text"
      },
      "last_name": {
        "type": "text"
      },
      "city": {
        "type": "text"
      },
      "zip": {
        "type": "keyword"
      },
      "state": {
        "type": "keyword"
      }
    }
  }
}


PUT /_enrich/policy/my-policy
{
  "match": {
    "indices": "users",
    "match_field": "email",
    "enrich_fields": ["first_name", "last_name", "city", "zip", "state"]
  }
}
```
</details>

<details>
<summary>Screenshots empty states</summary>

// Permissions error
<img width="1507" alt="Screenshot 2023-08-16 at 13 16 41" src="https://github.com/elastic/kibana/assets/1191206/73d2364f-054c-4977-a882-91e930b72629">
// Loading state
<img width="1078" alt="Screenshot 2023-08-07 at 11 43 17" src="https://github.com/elastic/kibana/assets/1191206/6e6fc769-7c3c-4319-a1d7-15866532c71c">
// No data
<img width="1076" alt="Screenshot 2023-08-07 at 11 42 28" src="https://github.com/elastic/kibana/assets/1191206/83413bfd-2dcf-4cae-addf-6f9d479526ee">

</details>

<details>
<summary>Screenshots</summary>

<img width="1507" alt="Screenshot 2023-08-15 at 14 33 52" src="https://github.com/elastic/kibana/assets/1191206/5dba5ee7-ebe9-4634-9ab1-91333df9d801">
<img width="1509" alt="Screenshot 2023-08-15 at 14 35 33" src="https://github.com/elastic/kibana/assets/1191206/13b2f895-b3e1-44e5-aad4-b71ccda3514b">
<img width="1510" alt="Screenshot 2023-08-15 at 14 34 11" src="https://github.com/elastic/kibana/assets/1191206/8427e27d-3081-47dc-837d-9818c2ee85b7">


</details>
